test/cli.js

```diff
@@ -83,6 +83,15 @@ test('enabling long stack traces will provide detailed debug information', t =>
 	});
 });
 
+test('`AssertionError` should capture infinity stack trace', t => {
+	execCli('fixture/infinity-stack-trace.js', (err, stdout, stderr) => {
+		t.ok(err);
+		t.match(stderr, /const c/);
+		t.match(stderr, /const b/);
```

Comment:
If I run the fixture locally (`node ./cli.js test/fixture/infinity-stack-trace.js`) I get:

```
  1 failed

  [anonymous]

  /Users/mark/GitHub/avajs/ava/test/fixture/infinity-stack-trace.js:6

   5: test(t => {
   6:   const c = () => t.fail();
   7:   const b = () => c();

  Test failed via `t.fail()`

  c (test/fixture/infinity-stack-trace.js:6:20)
  b (test/fixture/infinity-stack-trace.js:7:18)
  a (test/fixture/infinity-stack-trace.js:8:18)
  Test.t [as fn] (test/fixture/infinity-stack-trace.js:10:2)
```

Right now you're matching the code snippet but not the stack trace. Try `t.match(stderr, /c \(.+?infinity-stack-trace\.js:6/)` and so forth for `b` and `a`. (Note that I haven't actually tried this regular expression, apologies in advance if it's wrong! 😄 )